### PR TITLE
Adds support for manual harvest button

### LINKED ISSUE #4672: add manual harvest button to admin app
## User Story

In order to manually start a harvest job for a harvest source, datagov wants a button to press in the admin app to do so. 

## Acceptance Criteria

[ACs should be clearly demoable/verifiable whenever possible. Try specifying them using [BDD](https://en.wikipedia.org/wiki/Behavior-driven_development#Behavioral_specifications).]

- [ ] GIVEN I have visited a harvest source in the Flask admin app
  WHEN I press the "run harvest" button on the harvest source page
  THEN a new job is created in the harvest db with the proper config necessary for a downstream script to launch a new CF task in our harvest runner.

## Background

- [gha polling ticket](https://github.com/GSA/data.gov/issues/4670)
- [admin app](https://github.com/GSA/data.gov/issues/4619)

## Security Considerations ([required](https://nvd.nist.gov/800-53/Rev4/control/CM-4))

[comment]: # "Our SSP says 'The Data.gov team ensures security implications are considered as part of the agile requirements refinement process by including a section in the issue template used as a basis for new work.' so please don't remove this section without care."
[Any security concerns that might be implicated in the change. "None" is OK, just be explicit here!]

## Sketch

[Notes or a checklist reflecting our understanding of the selected approach]

- [ ] Add button to harvest_source Flask template
- [ ] On button press, create a new job in HarvestDB

### PR DESCRIPTION
**Pull request by btylerburton**
# Pull Request

Resolves https://github.com/GSA/data.gov/issues/4672

## About

This PR adds new base templates to the Flask app, standardizes view and edit forms for harvest sources and organizations, adds buttons and routes to trigger edit, harvest and delete. It also updates the harvest job table with new enums for status and adds a migration script.

## PR TASKS

- [X] The actual code changes.
- [X] Tests written and passed.
- [X] Any changes to docs?
- [ ] Bumped version number in [pyproject.toml](https://github.com/GSA/datagov-harvesting-logic/blob/8078c5b9f015ca7fb8a142e4e4b2e22ad2fd49b1/pyproject.toml#L3) (also checked on [PyPi](https://pypi.org/project/ckanext-geodatagov/#history)).

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -110,11 +112,25 @@ If you followed the instructions for `CKAN load testing` and `Harvester testing`
 3. when there are database DDL changes, use following steps to generate migration scripts and update database:
 
     ```bash
-    docker compose db up
+    docker compose up -d db
+    docker compose run app flask db stamp head
```
Comment (btylerburton) [inline README.md]: Can someone confirm this is correct? I ran into the error below, but this allowed me to proceed.

`ERROR [flask_migrate] Error: Target database is not up to date.`

Comment (Jin-Sun-tts) [reply README.md]: we need to run 'flask db upgrade' before 'flask db stamp head' to make sure changes be applied to database. And the 'flask db stamp head' command is optional I think.

Comment (btylerburton) [reply README.md]: Thanks, Jin! Let me try that out on a new branch to confirm.

Comment (btylerburton) [reply README.md]: I can't replicate it. Maybe I don't need to add the migration and it was just something on my local? Will bring this question to sync.

**Review by Jin-Sun-tts — DISMISSED** [review]

**Comment by btylerburton** [comment]
Hey @Jin-Sun-tts just merged in main branch, can you give me another approval?

**Review by Jin-Sun-tts — APPROVED** [review]

**Pull request merged**